packages/react-native/android/src/main/java/com/bugsnag/android/BugsnagReactNative.kt

```diff
@@ -25,15 +27,35 @@ class BugsnagReactNative(reactContext: ReactApplicationContext) :
     private fun configure(): WritableMap {
         return try {
             val client = Bugsnag.getClient()
+            bridge = reactContext.getJSModule(RCTDeviceEventEmitter::class.java)
             logger = client.logger
             plugin = client.getPlugin(BugsnagReactNativePlugin::class.java)!!
+            plugin.registerForMessageEvents { emitEvent(it) }
             plugin.configure().toWritableMap()
         } catch (exc: Throwable) {
             logFailure("configure", exc)
             WritableNativeMap()
         }
     }
 
+    /**
+     * Serializes a MessageEvent into a WritableMap and sends it across the React Bridge
+     */
+    private fun emitEvent(event: MessageEvent) {
+        logger.d("Received MessageEvent: ${event.type}")
+
+        val map = Arguments.createMap()
+        map.putString("type", event.type)
+
+        when (event.type) {
+            "ContextUpdate" -> map.putString("data", event.data as String?)
+            "UserUpdate" -> map.putMap("data", Arguments.makeNativeMap(event.data as Map<String, Any?>?))
+            "MetadataUpdate" -> map.putMap("data", Arguments.makeNativeMap(event.data as Map<String, Any?>?))
```

Comment:
Is it worth considering enumerating these types?

Reply:
Extracted these types to constants